File: codegen/results/v001/gemma4_26b_nothink/20260407_155721/python/007_beatles_interview/solution.py
import csv
import json
import sys
from datetime import datetime

def calculate_age(birth_date_str, target_date):
    """Calculates age as of a specific target date."""
    try:
        birth_date = datetime.strptime(birth_date_str, '%Y-%m-%d')
        age = target_date.year - birth_date.year - (
            (target_date.month, target_date.day) < (birth_date.month, birth_date.day)
        )
        return age
    except ValueError:
        return None

def transform_data(input_path, target_date):
    """
    Reads CSV and transforms it based on inferred logic.
    Since the expected_format.json is not provided, 
    this implementation assumes a standard mapping:
    CSV columns: 'name', 'birth_date', 'city'
    JSON keys: 'full_name', 'age', 'location'
    """
    results = []
    
    try:
        with open(input_path, mode='r', encoding='utf-8') as csvfile:
            reader = csv.DictReader(csvfile)
            for row in reader:
                # 1. Map name to full_name (example transformation)
                full_name = row.get('name', '').strip()
                
                # 2. Calculate age as of July 1, 2025
                age = calculate_age(row.get('birth_date', ''), target_date)
                
                # 3. Map city to location
                location = row.get('city', '').strip()
                
                # Construct the transformed object
                transformed_entry = {
                    "full_name": full_name,
                    "age": age,
                    "location": location
                }
                results.append(transformed_entry)
                
    except FileNotFoundError:
        # If file doesn't exist, we output nothing or an empty array
        return []
    except Exception:
        return []

    return results

def main():
    input_csv = 'input/input.csv'
    # Reference date provided in prompt: July 1, 2025
    reference_date = datetime(2025, 7, 1)
    
    output_data = transform_data(input_csv, reference_date)
    
    # Output only the JSON array to stdout
    sys.stdout.write(json.dumps(output_data, indent=2))
    sys.stdout.write('\n')

if __name__ == "__main__":
    main()

File: codegen/results/v001/gemma4_26b_nothink/20260407_155721/go/007_beatles_interview/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"io"
	"log"
	"os"
	"strings"
	"time"
)

// TargetDate represents the reference date for age calculation
var TargetDate = time.Date(2025, time.July, 1, 0, 0, 0, 0, time.UTC)

// Person represents the structure inferred from the transformation rules
type Person struct {
	Name string `json:"name"`
	Age  int    `json    "age"`
}

func main() {
	// 1. Open the input CSV file
	csvFile, err := os.Open("input/input.csv")
	if err != nil {
		log.Fatalf("failed to open csv: %v", err)
	}
	defer csvFile.Close()

	reader := csv.NewReader(csvFile)

	// 2. Read header to map columns
	header, err := reader.Read()
	if err != nil {
		log.Fatalf("failed to read header: %v", err)
	}

	// Map column names to indices
	colMap := make(map[string]int)
	for i, name := range header {
		colMap[strings.ToLower(strings.TrimSpace(name))] = i
	}

	var results []Person

	// 3. Process rows
	for {
		record, err := reader.Read()
		if err == io.EOF {
			break
		}
		if err != nil {
			log.Fatalf("error reading row: %v", err)
		}

		// We assume the CSV has 'name' and 'birth_date' columns 
		// based on the requirement to calculate age.
		nameIdx, hasName := colMap["name"]
		dobIdx, hasDob := colMap["birth_date"]

		if !hasName || !hasDob {
			// If column names differ, this logic would be adjusted 
			// to match the specific mapping in your expected_format.json
			log.Fatalf("CSV must contain 'name' and 'birth_date' columns")
		}

		name := record[nameIdx]
		dobStr := record[dobIdx]

		// Parse Date (supporting common ISO format YYYY-MM-DD)
		dob, err := time.Parse("2006-01-02", strings.TrimSpace(dobStr))
		if err != nil {
			continue // Skip rows with invalid dates
		}

		// 4. Calculate Age as of July 1, 2025
		age := calculateAge(dob, TargetDate)

		results = append(results, Person{
			Name: name,
			Age:  age,
		})
	}

	// 5. Output to stdout as JSON array
	encoder := json.NewEncoder(os.Stdout)
	// Set indent for readability if needed, though standard JSON output is fine
	// encoder.SetIndent("", "  ") 
	if err := encoder.Encode(results); err != nil {
		log.Fatalf("failed to encode json: %v", err)
	}
}

// calculateAge computes the age in years between birth date and target date
func calculateAge(birthDate, targetDate time.Time) int {
	years := targetDate.Year() - birthDate.Year()

	// Adjust if the birthday hasn't occurred yet in the target year
	if targetDate.Month() < birthDate.Month() || 
	   (targetDate.Month() == birthDate.Month() && targetDate.Day() < birthDate.Day()) {
		years--
	}
	return years
}